WalletWasabi.Gui/TorOnlyEndpointConnector.cs

```diff
@@ -0,0 +1,34 @@
+#if !NOSOCKET
+using NBitcoin;
+using NBitcoin.Protocol;
+using NBitcoin.Protocol.Behaviors;
+using NBitcoin.Protocol.Connectors;
+using NBitcoin.Socks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Gui
+{
+	public class TorOnlyEndpointConnector : IEnpointConnector
+	{
+		private DefaultEndpointConnector _defaultEndpointConnector = new DefaultEndpointConnector(); 
+
+		public TorOnlyEndpointConnector()
+		{
+		}
+
+		public async Task ConnectSocket(Socket socket, EndPoint endpoint, NodeConnectionParameters nodeConnectionParameters, CancellationToken cancellationToken)
+		{
+			if(!endpoint.IsTor()) 
```

Comment:
you should better throw an exception, the code after this expect the socket to be open if no exception

Reply:
Another idea is to code that directly in the `DefaultEndpointConnecter` a `AllowOnlyTorEndpoints`.